src/main/java/com/alipay/oceanbase/rpc/table/ObTable.java

```diff
@@ -394,8 +391,28 @@ public ObPayload execute(final ObPayload request) throws RemotingException,
         } catch (Exception ex) {
             throw new ObTableConnectionStatusException("check status failed", ex);
         }
-
-        return realClient.invokeSync(connection, request, obTableExecuteTimeout);
+        boolean needReconnect = false;
+        int retryTimes = 0;
+        ObPayload payload = null;
+        do {
+            retryTimes++;
+            try {
+                if (needReconnect) {
+                    String msg = String.format("Receive error: tenant not in server and reconnect it, ip:{}, port:{}, tenant id:{}, retryTimes: {}",
+                            connection.getObTable().getIp(), connection.getObTable().getPort(), connection.getTenantId(), retryTimes);
+                    connection.reConnectAndLogin(msg);
+                    needReconnect = false;
+                }
+                payload = realClient.invokeSync(connection, request, obTableExecuteTimeout);
+            } catch (ObTableException ex) {
+                if (ex instanceof ObTableTenantNotInServerException && retryTimes < 2) {
```

Comment:
It seems that we don't need ```retryTimes < 2```, since if retryTimes is bigger than 1, we exist the while clause.

Reply:
We need it. When retryTimes == 2, it means it has receive the `ObTableTenantNotInServerException` twice and need to throw this Exception to the upper function